src/hssm/hssm.py

```diff
@@ -510,24 +511,41 @@ def sample(
                     + "The jitter argument will be ignored."
                 )
                 del kwargs["jitter"]
-            else:
-                pass
 
-        self._inference_obj = self.model.fit(
-            inference_method=sampler, init=init, **kwargs
-        )
+        if "include_mean" not in kwargs:
+            # If not specified, include the mean prediction in
+            # kwargs to be passed to the model.fit() method
+            kwargs["include_mean"] = True
+        idata = self.model.fit(inference_method=sampler, init=init, **kwargs)
+
+        if self._inference_obj is None:
+            self._inference_obj = idata
+        elif isinstance(self._inference_obj, az.InferenceData):
+            self._inference_obj.extend(idata)
+        else:
+            raise ValueError(
+                "The model has an attached inference object under"
+                + " self._inference_obj, but it is not an InferenceData object."
+            )
 
         # The parent was previously not part of deterministics --> compute it via
         # posterior_predictive (works because it acts as the 'mu' parameter
         # in the GLM as far as bambi is concerned)
         if self._inference_obj is not None:
             if self._parent not in self._inference_obj.posterior.data_vars.keys():
-                self.model.predict(self._inference_obj, kind="mean", inplace=True)
+                # self.model.predict(self._inference_obj, kind="mean", inplace=True)
                 # rename 'rt,response_mean' to 'v' so in the traces everything
                 # looks the way it should
                 self._inference_obj.rename_vars(
                     {"rt,response_mean": self._parent}, inplace=True
                 )
+            elif (
+                self._parent in self._inference_obj.posterior.data_vars.keys()
+                and "rt,response_mean" in self._inference_obj.posterior.data_vars.keys()
```

Comment:
`data_vars` are `dicts`, so the Python 3 style is to not use `keys()`

Reply:
changed